ant-tech-alliance/ray
# Support building Java and Python version at the same time.

## What do these changes do?
Support building Java and Python version at the same time.

This is the required feature for hybrid (Java & Python) cluster.

How to test:
1. Change [this line](https://github.com/ray-project/ray/blob/master/python/setup.py#L76) from `subprocess.check_call(["../build.sh", "-p", sys.executable])` to `subprocess.check_call(["../build.sh", "-l", "all"])`.
2. `pip install -e python/ --verbose`
3. `python test/runtest.py`
4. `cd java`
5. `mvn clean install -Dmaven.test.skip`
6. Before running `mvn test`, make sure there is no ray related processes.
7. `mvn test`


## Related issue number
